app/controllers/admin/attachments_controller.rb

```diff
@@ -25,6 +28,10 @@ def create
     end
   end
 
+  def edit
+    render(preview_design_system_user? ? "edit" : "edit_legacy")
```

Comment:
If the fail is unsuccessful it will rerender the edit view here https://github.com/alphagov/whitehall/blob/87af8214f730585c3273613ee4a31f5ad26a06f6/app/controllers/admin/attachments_controller.rb#L44-L46

We will also need this logic 
```
render(preview_design_system_user? ? "edit" : "edit_legacy")
```

there so the correct page is rerendered